src/plugins/ui_stateful/selection.ts

```diff
@@ -422,13 +422,23 @@ export class GridSelectionPlugin extends UIPlugin {
   }
 
   getStatisticFnResults(): { [name: string]: number | undefined } {
-    // get deduplicated cells in zones
-    const cells = new Set(
-      this.gridSelection.zones
-        .map((zone) => this.getters.getEvaluatedCellsInZone(this.getters.getActiveSheetId(), zone))
-        .flat()
-        .filter((cell) => cell.type !== CellValueType.empty)
-    );
+    const sheetId = this.getters.getActiveSheetId();
+    const cells = new Set<EvaluatedCell>();
+
+    for (const zone of this.gridSelection.zones) {
+      for (let col = zone.left; col <= zone.right; col++) {
+        for (let row = zone.top; row <= zone.bottom; row++) {
```

Comment:
minor suggestion to avoid one nested level and with a more declarative approach.
LGTM otherwise :)
```suggestion
      for (const { col, row } of positions(zone)) {
```

Reply:
@LucasLefevre, I have implemented the suggested changes. Thanks :)